test/libsolidity/SolidityEndToEndTest.cpp

```diff
@@ -9077,6 +9077,25 @@ BOOST_AUTO_TEST_CASE(invalid_instruction)
 	BOOST_CHECK(callContractFunction("f()") == encodeArgs());
 }
 
+BOOST_AUTO_TEST_CASE(revert)
+{
+	char const* sourceCode = R"(
+		contract C {
+			function f() {
+				revert();
```

Comment:
Please perform state changes here and check that they are not applied. Otherwise, the test would also succeed for the case where `revert` does nothing.